src/games/strategy/engine/framework/startup/launcher/LocalLauncher.java

```diff
@@ -38,7 +39,7 @@ protected void launchInNewThread(final Component parent) {
     ServerGame game = null;
     try {
       m_gameData.doPreGameStartDataModifications(m_playerListing);
-      final IServerMessenger messenger = new DummyMessenger();
+      final IServerMessenger messenger = mock(IServerMessenger.class);
```

Comment:
A `DummyMessenger` is a 'do-nothing' object. That is different from a mock.
The need for a `DummyMessenger` is actually a code smell. It indicates a lot of code that should not be called, but is. Secondly, it looks to be an inappropriate abstraction. It is overly broad and requires dependencies that are not always used.